### PR TITLE
Unify docs actions and automate release documentation generation

### PR DESCRIPTION
**Pull request by rasolca**

### CODE REVIEW

File: .github/workflows/doc.yml [hunk 1]
```diff
@@ -18,4 +23,22 @@ jobs:
...
-          cd doc && make doc
+          cd doc && make doc && cd ..
+      - name: Push Docs master
+        if: "github.repository == 'eth-cscs/DLA-Future' && github.ref == 'refs/heads/master'"
+        run: |
+          mv doc/html master
+          COMMIT_MSG=`git log -n 1 --pretty=format:%s`
+          git config user.name "Doc CI Action" && git config user.email "rasolca@users.noreply.github.com"
+          git fetch --no-tags --depth=1 origin +refs/heads/docs:refs/heads/docs
+          git symbolic-ref HEAD refs/heads/docs && git reset
```
Comment (RMeli) [inline .github/workflows/doc.yml:34]: Nitpick: can these commands be lifted into their own step, since they are shared?

**Review by RMeli — APPROVED** [review]
LGTM

**Review by albestro — APPROVED** [review]
LGTM

File: .github/workflows/doc.yml (hunk 1 above)
Comment (rasolca) [reply .github/workflows/doc.yml:34]: I thought about it, but this IMO this solution was the cleaner.
As for master I need to get the commit message before switching the branch the if clauses become a mess.

Comment (RMeli) [reply .github/workflows/doc.yml:34]: Fair. I think the `if` clause could simply define the `TAG` (either the actual tag or master) and `COMMIT_MSG` variables, and then everything else would be the same? But more work than it's worth, so happy to leave it as-is.

**Pull request merged**